### PR TITLE
Include xref reference to dependency map

### PR DESCRIPTION
**Pull request by live1206**
#3635 
- Include `xref` reference to the dependency map
- Update tests to verify `build.manifest`
- dependency type should always be `UidInclusion` since `@uid` implicitly means `@uid?displayProperty=name`

### CODE REVIEW

**Comment by superyyrrzz** [comment]
> dependency type is Link while using @uid in .md or reference to uid in SDP
In markdown, `@uid` implicitly means `@uid?displayProperty=name`, so it's dependency type should be `UidInclusion`?

File: src/docfx/build/page/AttributeTransformer.cs [hunk 1]
```diff
@@ -32,13 +32,15 @@ object TransformContent(IEnumerable<DataTypeAttribute> attributes, object value,
                     var (html, markup) = Markup.ToHtml((string)value, file, ReadFileDelegate, GetLinkDelegate, ResolveXrefDelegate, null, MarkdownPipelineType.Markdown);
                     errors.AddRange(markup.Errors);
                     result = html;
+                    XrefMap.HandleDependencyMap(markup, file, callback?.DependencyMapBuilder);
```
Comment (OsmondJiang) [inline src/docfx/build/page/AttributeTransformer.cs]: >callback [](start = 62, length = 8)

this callback is really confused for me

Comment (live1206) [reply src/docfx/build/page/AttributeTransformer.cs]: I can expand it

Comment (OsmondJiang) [reply src/docfx/build/page/AttributeTransformer.cs]: no hurry, let's together find a easy and simple way

---
In reply to: [235588157](https://github.com/dotnet/docfx/pull/3646#discussion_r235588157) [](ancestors = 235588157)

File: src/docfx/build/page/AttributeTransformer.cs [hunk 2]
```diff
@@ -32,13 +32,15 @@ object TransformContent(IEnumerable<DataTypeAttribute> attributes, object value,
...
                 }
 
                 if (attribute is InlineMarkdownAttribute)
                 {
                     var (html, markup) = Markup.ToHtml((string)value, file, ReadFileDelegate, GetLinkDelegate, ResolveXrefDelegate, null, MarkdownPipelineType.InlineMarkdown);
                     errors.AddRange(markup.Errors);
                     result = html;
+                    XrefMap.HandleDependencyMap(markup, file, callback?.DependencyMapBuilder);
```
Comment (OsmondJiang) [inline src/docfx/build/page/AttributeTransformer.cs]: >XrefMap.HandleDependencyMap(markup, file, callback?.DependencyMapBuilder); [](start = 20, length = 74)

how about put it inside of Markup?

File: src/docfx/build/xref/XrefMap.cs [hunk 3]
```diff
@@ -81,6 +87,14 @@ public XrefSpec Resolve(string uid, string moniker = null)
                 return externalSpec;
             }
             return null;
+
+            void HandleDependencyMap(XrefSpec spec)
```
Comment (yufeih) [inline src/docfx/build/xref/XrefMap.cs]: just call `AddDependencyItem`

File: src/docfx/build/xref/XrefMap.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ public IEnumerable<XrefSpec> InternalReferences
             }
         }
 
-        public XrefSpec Resolve(string uid, string moniker = null)
+        public XrefSpec Resolve(string uid, Document file, DependencyMapBuilder dependencyMapBuilder, string moniker = null)
```
Comment (yufeih) [inline src/docfx/build/xref/XrefMap.cs]: move this to upper layer

File: src/docfx/build/xref/XRefSpec.cs [hunk 5]
```diff
@@ -19,6 +19,9 @@ public class XrefSpec
         [JsonExtensionData]
         public JObject ExtensionData { get; } = new JObject();
 
+        [JsonIgnore]
```
Comment (yufeih) [inline src/docfx/build/xref/XRefSpec.cs]: Make xrefmap resolve return `(XrefSpec, Document?)`

**Pull request merged**